pkg/controller/cluster/resource/resource.go

```diff
@@ -97,7 +97,11 @@ func StatefulSetForRack(r scyllav1alpha1.RackSpec, c *scyllav1alpha1.Cluster, si
 	if placement == nil {
 		placement = &scyllav1alpha1.PlacementSpec{}
 	}
-
+	opt := true
+	scyllaConfigMap := r.ScyllaConfig
+	if scyllaConfigMap == "" {
+		scyllaConfigMap = "scylla-config"
```

Comment:
I think it's better to be explicit here and not have a default value assumed inside the operator.
I know that the operator does exactly this thing in other places (eg image name) but in my experience it's preferable to be explicit.
It's better to use [MutatingWebhooks](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#mutatingadmissionwebhook) for this job.

The problem here is that if the user creates a cluster without a ConfigMap, then they can't include one, unless we add the logic for it.

My suggestion would be to leave a TODO here to stop assuming a default ConfigMap once we implement the ConfigMap change action in the operator. 
What do you think?

Reply:
I am not sure I understand. If we create one here then they can reuse that later if they want. If we don't set one at all then they can't add one.

Or have I misunderstood? The mount later isn't mandatory so a non existing config map will just wait for it to be mounted later if needed.

Reply:
Afact the only thing they can't do as it stands now is to change the config map name but they can skip or set another initially.

Reply:
We can use a validating webhook to enforce it?

Reply:
I agree we should leave a TODO for now and implement a change function separately.